connection.go

```diff
@@ -329,6 +340,16 @@ func (sc *snowflakeConn) ExecContext(ctx context.Context, query string, args []d
 }
 
 func (sc *snowflakeConn) QueryContext(ctx context.Context, query string, args []driver.NamedValue) (driver.Rows, error) {
+	qid, _ := getResumeQueryID(ctx)
+	if qid == "" {
+		return sc.queryContextInternal(ctx, query, args)
+	}
+	actualQuery := fmt.Sprintf("select * from table(result_scan('%s'))", qid)
```

Comment:
Probably we can assume that `qid` will be a safe value, but maybe it would be good to avoid SQL injection by for example doing a basic regex validation?

Reply:
The driver is submitting a query on user's behavior, and they can explore SQL injection in UI too if they want to. So we cannot protect user's own data from user's own misbehavior. The security question we snowflake want to answer is, will there be any SQL that could escape the authorization mechanism we have when there is a malicious user? I think that is not addressed on client side anyway.

Reply:
Agreed that we should protect against injection.

> The driver is submitting a query on user's behavior, and they can explore SQL injection in UI too if they want to.

The difference here is that the user doesn't have a chance to sanitize the data themselves. The correct way to sanitize it is a bind variable which the user can't do. We can also just verify that the value is a UUID -- why not?

Reply:
Users hurting themselves is one thing (we still should protect them as long as we can) but men in the middle is another thing. I'm not that too familiar with `gosnowflake` so I can be wrong,  but plain queries always makes me uncomfortable.